orte/tools/orte-info/Makefile.am

```diff
@@ -26,7 +26,7 @@ AM_CFLAGS = \
             -DOPAL_CONFIGURE_DATE="\"@OPAL_CONFIGURE_DATE@\"" \
             -DOMPI_BUILD_USER="\"$$USER\"" \
             -DOMPI_BUILD_HOST="\"`(hostname || uname -n) | sed 1q`\"" \
```

Comment:
Tested it and this line still caused variations.
=>
```
\"$${HOSTNAME:-`(hostname || uname -n) 2> /dev/null | sed 1q`}\"" \
```

Reply:
oshmem_info and mpisync will therefore cause the same issue.

Reply:
So we need that line to change? FWIW: that suggested line is what we have in PRRTE today (someone asked for reproducible build over there earlier)

Reply:
I just pushed that change here (and #8148) for orte-info and mpisync.  Anywhere you have a `-DBUILD_HOST` kind of construct in PMIX/PRTE Makefile.am's, you'll want to allow the `$HOSTNAME` override.